File: playframework-10937-body-redirect-to-server/app/controllers/HomeController.scala
package controllers

import akka.util.ByteString

import javax.inject._
import play.api.mvc._

@Singleton
class HomeController @Inject()(val controllerComponents: ControllerComponents) extends BaseController {

  def index() = Action { implicit request: Request[AnyContent] =>
    println("Headers:")
    println("--------")
    request.headers.headers.foreach(header => println(s"${header._1}: ${header._2}"))
    println()
    println("Body size: " + request.body.asRaw.size)
    println()
    println("Body:")
    println("-----")
    println(request.body.asRaw.flatMap(_.asBytes()).getOrElse(ByteString.emptyByteString).utf8String)
    println()
    Ok("hello world!")
  }
}

File: playframework-11220/app/controllers/HomeController.java
package controllers;

import play.data.FormFactory;
import play.i18n.Lang;
import play.libs.typedmap.TypedMap;
import play.mvc.*;

import javax.inject.Inject;
import java.util.HashMap;
import java.util.Map;

public class HomeController extends Controller {

    private final FormFactory formFactory;

    @Inject
    public HomeController(final FormFactory formFactory) {
        this.formFactory = formFactory;
    }

    public Result index() {
        this.formFactory.form().bind(Lang.defaultLang(), TypedMap.create(), new HashMap<>());
        return ok("hello world");
    }

}
